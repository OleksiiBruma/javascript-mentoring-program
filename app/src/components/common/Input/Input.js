import React from "react";
import PropTypes from "prop-types"
import { TextField } from "@material-ui/core";

const CustomInput = ({
  inputData: { name, error, value, errorText, type, multiline, rows, variant },
  handleBlur,
  handleChange
}) => (
  <TextField
    key={name}
    error={error}
    label={name}
    name={name}
    type={type}
    fullWidth
    multiline={multiline}
    rows={rows}
    value={value}
    onChange={handleChange}
    onBlur={handleBlur}
    variant={variant}
    helperText={error && errorText}
    margin="normal"
  />
);

CustomInput.propTypes = {
  inputData: PropTypes.shape(
    {
      name: PropTypes.string.isRequired,
      error: PropTypes.bool.isRequired,
      errorText: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired,
      value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
        PropTypes.instanceOf(Date)
      ])
    }
  ),
  handleChange : PropTypes.func.isRequired,
  handleBlur : PropTypes.func.isRequired,
}



export default CustomInput;
