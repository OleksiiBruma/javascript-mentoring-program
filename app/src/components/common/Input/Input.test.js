import renderer from "react-test-renderer";
import React from "react";
import Input from "../Input";

it("Input ", () => {
  const inputData = {
    name: "login",
    value: "",
    error: false,
    emptyText: `required`,
    incorrectText: `incorrect login`,
    errorText: ``,
    type: "text"
  };
  const handlerChange = () => {};
  const handleBlur = () => {};

  const component = renderer.create(
    <Input inputData={inputData} handleChange={handlerChange}
    handleBlur={handleBlur} />
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
