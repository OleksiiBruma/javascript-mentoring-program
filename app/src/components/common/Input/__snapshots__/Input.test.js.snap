// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input  1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
  >
    login
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      name="login"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
  </div>
</div>
`;
